prql-compiler/src/semantic/transforms.rs

```diff
@@ -847,29 +875,113 @@ impl AstFold for Flattener {
     }
 }
 
-// TODO: Can we dynamically get the types, like in pandas? We need to put
-// quotes around strings and not around numbers.
-// https://stackoverflow.com/questions/64369887/how-do-i-read-csv-data-without-knowing-the-structure-at-compile-time
-fn parse_csv(text: &str) -> Result<RelationLiteral> {
-    let mut rdr = csv::Reader::from_reader(text.as_bytes());
+mod from_text {
+    use super::*;
 
-    Ok(RelationLiteral {
-        columns: rdr
-            .headers()?
-            .into_iter()
-            .map(|h| h.to_string())
-            .collect::<Vec<_>>(),
-        rows: rdr
-            .records()
-            .into_iter()
-            // This is messy rust, but I can't get it to resolve the Errors
-            // when it leads with `row_result?`. I'm sure it's possible...
-            .map(|row_result| {
-                row_result.map(|row| row.into_iter().map(|x| x.to_string()).collect())
+    // TODO: Can we dynamically get the types, like in pandas? We need to put
+    // quotes around strings and not around numbers.
+    // https://stackoverflow.com/questions/64369887/how-do-i-read-csv-data-without-knowing-the-structure-at-compile-time
```

Comment:
I actually think this wasn't a problem in the end?

Reply:
I just used `Literal` as a cell type so I can contain basically anything PRQL supports.

Reply:
Perfect!